src/swsssdk/dbconnector.py

```diff
@@ -26,165 +29,384 @@ def load_sonic_db_config(sonic_db_file_path=SONIC_DB_CONFIG_FILE):
+        return 0
 
-    def delete(self, db_name, key, *args, **kwargs):
-        db_id = self.get_dbid(db_name)
-        return super(SonicV2Connector, self).delete(db_id, key, *args, **kwargs)
+    def keys(self, db_name, pattern='*', namespace='none', *args, **kwargs):
+        if namespace == 'none':
+            namespace = self.ns_name
+
+        data = []
+        ns_instances = []
+        if namespace == 'all':
+            ns_instances = self.get_db_connection_instances()
+        else:
+            ns_instances.append(namespace)
+
+        for instance in ns_instances:
+            keys = []
+            db_id = self.get_dbid(db_name, instance)
+            keys = self.db_intfs[instance].keys(db_id, pattern, *args, **kwargs)
+            if keys != None:
+                data.extend(keys)
+
+        return data
+
+    def get(self, db_name, _hash, key, namespace='none', *args, **kwargs):
+        # If the argument is 'all' get will return the vale of first match
+        if namespace == 'none':
+            namespace = self.ns_name
+
+        ns_instances = []
+        if namespace == 'all':
+            ns_instances = self.get_db_connection_instances()
+        else:
+            ns_instances.append(namespace)
+
+        for instance in ns_instances:
+            db_id = self.get_dbid(db_name, namespace)
+            data = self.db_intfs[namespace].get(db_id, _hash, key, *args, **kwargs)
+            if data:
+                return data
+
+        # We reach here, the key is not present
+        return None 
+
+    # This function will get the keys and do a get_all per namespace.
+    def get_keys_and_all(self, db_name, _hash, namespace='none', *args, **kwargs):
+        if namespace == 'none':
+            namespace = self.ns_name
+
+        got_data = []
+        ns_instances = []
+        if namespace == 'all':
+            ns_instances = self.get_db_connection_instances()
+        else:
+            ns_instances.append(namespace)
+
+        for instance in ns_instances:
+            db_id = self.get_dbid(db_name, instance)
+            keys = self.db_intfs[instance].keys(db_id, _hash, *args, **kwargs)
+            if keys:
+                for entry in keys:
+                    key = entry.decode()
+                    data = self.db_intfs[instance].get_all(db_id, key, *args, **kwargs)
+                    if data:
+                        got_data.extend(data)
+
+        return got_data
+
+    def get_all(self, db_name, _hash, namespace='none', *args, **kwargs):
+        if namespace == 'all':
+            return get_keys_and_all(db_name, _hash, namespace, *args, **kwargs)
+
+        if namespace == 'none':
+            namespace = self.ns_name
+
+        db_id = self.get_dbid(db_name, namespace)
+        return self.db_intfs[namespace].get_all(db_id, _hash, *args, **kwargs)
+
+    def set(self, db_name, _hash, key, val, namespace='none', *args, **kwargs):
+        if namespace == 'none':
```

Comment:
>namespace == 'none' [](start = 11, length = 19)

`none` is a valid namespace name. you may use `if namespace:`

Reply:
Updated the code and used None keyword instead.